gems/smithy-client/lib/smithy-client/protocols/rpc_v2.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    module Protocols
+      # RPCV2 protocol
+      # TODO:
+      #  * Refactor methods to handle eventstreams
+      #  * Decide whether we should recognize query traits here
+      #  * Add Documentation
+      class RPCv2
+        SHAPE_ID = 'smithy.protocols#rpcv2Cbor'
+
+        def initialize(options = {})
+          @query_compatible = options[:query_compatible]
```

Comment:
This "query-compatible" is very aws-specific but I want to demo how we could have "settings" on protocol.

I imagined the settings idea will be more relevant for other protocols like JSON (customer decides to whether to honor `jsonName` trait).

That being said - the "home" for RPCV2 protocol is in `smithy-client` gem and we need to figure out a way to recognize `query_compatible` trait when we use this protocol for AWS services. 

We could just recognize the query trait here but that goes against the grain of separating between generic and aws. Alternatively, maybe `aws-sdk-core` could have a RPCV2 protocol class that inherits this and layers "query-compatible" code a top of it? Open to any thoughts.

Reply:
Yeah, this is a tricky line between generic and AWS that rpcv2 blurs.  We never quite dealt with it in the java smithy-ruby (we just didn't implement the query compatible part).

I'd honestly lean towards leaving the setting and check against the trait in here.  While it does check against an AWS defined trait, its using only the string shape ID and doesn't bring in any additional dependencies (at codegen or runtime).